xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ForgotUsername.xml

```diff
@@ -49,6 +49,10 @@
 
 #else
   #set($results = $services.query.hql(", BaseObject obj, StringProperty prop where obj.name = doc.fullName and obj.className = 'XWiki.XWikiUsers' and prop.id.id = obj.id and prop.id.name = 'email' and LOWER(prop.value) = ?").bindValues([$email.toLowerCase()]).execute())
+  ## If local user does not exist check global user
+  #if($results.size() == 0)
+    #set($results = $services.query.hql(", BaseObject obj, StringProperty prop where obj.name = doc.fullName and obj.className = 'XWiki.XWikiUsers' and prop.id.id = obj.id and prop.id.name = 'email' and LOWER(prop.value) = ?").bindValues([$email.toLowerCase()]).setWiki('xwiki').execute())
```

Comment:
2 issues to avoid:
* Code duplication of the query that is created in the line above. You should reuse the query generated in the above line and execute it again, but for a different (main wiki)
* Hardcoding things (like the main wiki's name, `xwiki`). Instead, we should obtain it from `$xcontext.mainWikiName`